src/main/scala/com/campudus/tableaux/router/auth/permission.scala

```diff
@@ -0,0 +1,305 @@
+package com.campudus.tableaux.router.auth
+
+import com.campudus.tableaux.UnauthorizedException
+import com.campudus.tableaux.database.domain.{ColumnType, Table}
+import com.campudus.tableaux.helper.JsonUtils.asSeqOf
+import org.vertx.scala.core.json._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+object RoleModel {
+
+  def apply(jsonObject: JsonObject): RoleModel = {
+    new RoleModel(jsonObject)
+  }
+}
+
+case class RoleModel(jsonObject: JsonObject) {
+
+  /**
+    * Checks the RoleModel if a specific action on a scope is allowed for a set of given roles.
+    * If not a ForbiddenException is thrown
+    *
+    *  There are several steps
+    *    1. filter permissions for assigned user roles, desired action and scope
+    *    2. first check if a permission with type 'grant' meets the conditions
+    *        no condition means -> condition evaluates to 'true'
+    *
+    *    3. if at least one meets the conditions check also the deny types
+    *
+    * @return
+    */
+  def checkAuthorization(
+      requestRoles: Seq[String],
+      action: Action,
+      scope: Scope,
+      tableOpt: Option[Table]
+  ): Future[Unit] = {
+
+    val grantPermissions: Seq[Permission] = filterPermissions(requestRoles, Grant, action, scope)
+
+    val isAllowed: Boolean = grantPermissions.exists(_.isMatching(tableOpt))
+
+//    Console.println(s"XXX $isAllowed")
+
+    if (!isAllowed) {
+      return Future.failed(UnauthorizedException(action, scope))
+    }
+
+    // simples test working
+    // TODO test with mixed grant and deny types
+
+    if (isAllowed) {
+      val denyPermissions: Seq[Permission] = filterPermissions(requestRoles, Deny, action, scope)
+      val isDenied: Boolean = denyPermissions.exists(_.isMatching(tableOpt))
+
+      if (isDenied) {
+        return Future.failed(UnauthorizedException(action, scope))
+      }
+    }
+
+    Future.successful(())
+  }
+
+  val role2permissions: Map[String, Seq[Permission]] =
+    jsonObject
+      .fieldNames()
+      .asScala
+      .map(
+        key => {
+          val permissionsJson: Seq[JsonObject] = asSeqOf[JsonObject](jsonObject.getJsonArray(key))
+          (key, permissionsJson.map(permissionJson => Permission(permissionJson)))
+        }
+      )
+      .toMap
+
+  override def toString: String =
+    role2permissions
+      .map({
+        case (key, permission) => s"$key => ${permission.toString}"
+      })
+      .mkString("\n")
+
+  // TODO possibly obsolete
+  def getPermissionsForRoles(roleNames: Seq[String]): Seq[Permission] =
+    role2permissions.filter({ case (k, _) => roleNames.contains(k) }).values.flatten.toSeq
```

Comment:
Please don't use this short variable names. It's not readable for other people. Please name them like `key`